File: 2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.js
/**
 * @param {string} s
 * @return {number}
 */
var minLength = function(s) {
    const stack = [s[0]]
    const map = new Set(["AB", "CD"])

    for (let i = 1; i < s.length; i++) {
        if(map.has(stack[stack.length - 1] + s[i])){
            stack.pop()
        }  else {
            stack.push(s[i])
        }
    }

    console.log(stack)
    return stack.length
};

File: 2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.py
class Solution:
    def minLength(self, s: str) -> int:
        m = set(["AB", "CD"])

        stack = []

        for char in s:
            if not stack:
                stack.append(char)
                continue
            
            print(stack[-1] + char)
            if (stack[-1] + char) in m:
                stack.pop()
            else:
                stack.append(char)
        
        return len(stack)